### PR TITLE
Implement a way to compose vNotes

### LINKED ISSUE #10: Feature: Compose and send notes
[Debian #549274](http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=549274)

### PR DESCRIPTION
**Pull request by cschramm**
Closes #10

### CODE REVIEW

File: blueman/plugins/manager/Notes.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    data = ('BEGIN:VNOTE \n'
+            'VERSION:1.1 \n'
+            'BODY;CHARSET=UTF-8: %s \n'
+            'DCREATED:%s \n'
+            'LAST-MODIFIED:%s \n'
+            'CLASS:PUBLIC \n'
+            'X-IRMC-LUID:000001000000 \n'
+            'END:VNOTE \n' % (' '.join(text.splitlines()), date, date))
+
+    tempfile = NamedTemporaryFile(delete=False)
```
Comment (infirit) [inline blueman/plugins/manager/Notes.py]: Not an issue but the ``tempfile`` module supports the ``with`` statement, which is convenient as it cleans up after itself :smile: .

```python
with NamedTemporaryile(delete=False) as tempfile:
    do_stuff
```

Comment (cschramm) [inline blueman/plugins/manager/Notes.py]: I don't think that's applicable here as I need to close and unlink separately (before and after running blueman-sendto), is it?

Comment (infirit) [inline blueman/plugins/manager/Notes.py]: Wait, ``launch`` does not block so by the time ``blueman-sendto`` is started the file is already unlinked. At least this is what happens on my machine when I tested this. We probably either need to wait for ``blueman-sendto`` to finish (I see no way with launch) or initiate the transfer through the bluez classes and have the callback clean things of on error and success.

**Comment by cschramm** [comment]
I think I found a nice solution now adding a new feature to blueman-sendto as a side product. :smile:

**Comment by infirit** [comment]
After some fixes from my moving things around it does not work for me. See below the applet output. Normal files work fine, is there something I need to do on the phone (Galaxy nexus with custom rom)?

Also, you will want to set the ``activates-default`` on the GtkEntry and then on the ok GtkButton set ``can-default`` so that the enter key will trigger the ok button, at least that is what should happen :smile:.

```
do_g_properties_changed (/usr/lib64/python2.7/site-packages/blueman/bluez/obex/Transfer.py:29)
/org/bluez/obex/client/session1/transfer11 Filename /tmp/noteIJHTUL.vnt 
_________
do_g_properties_changed (/usr/lib64/python2.7/site-packages/blueman/bluez/obex/Transfer.py:29)
/org/bluez/obex/client/session1/transfer11 Session /org/bluez/obex/client/session1 
_________
do_g_properties_changed (/usr/lib64/python2.7/site-packages/blueman/bluez/obex/Transfer.py:29)
/org/bluez/obex/client/session1/transfer11 Size 168 
_________
do_object_added (/usr/lib64/python2.7/site-packages/blueman/bluez/obex/Manager.py:49)
/org/bluez/obex/client/session1/transfer11 
_________
do_g_properties_changed (/usr/lib64/python2.7/site-packages/blueman/bluez/obex/Transfer.py:29)
/org/bluez/obex/client/session1/transfer11 Status error 
_________
_on_transfer_completed (/usr/lib64/python2.7/site-packages/blueman/bluez/obex/Manager.py:70)
/org/bluez/obex/client/session1/transfer11 False 
```

**Comment by cschramm** [comment]
I improved the dialog.

Android does not support vNotes. There are some applications for Android that do and might enable the phone to receive notes. Unfortunately I do not know of a way to detect if vNotes are supported or not (other than trying to send one and fail or succeed).

**Pull request closed without merging**